src/Compilers/Core/Portable/Diagnostic/ProgrammaticSuppressionInfo.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Contains information about the source of a programmatic diagnostic suppression produced by an <see cref="DiagnosticSuppressor"/>.
+    /// </summary>
+    internal sealed class ProgrammaticSuppressionInfo : IEquatable<ProgrammaticSuppressionInfo>
+    {
+        public ImmutableHashSet<(string Id, LocalizableString Justification)> Suppressions { get; }
+
+        internal ProgrammaticSuppressionInfo(ImmutableHashSet<(string Id, LocalizableString Justification)> suppressions)
+        {
+            Suppressions = suppressions;
+        }
+
+        public bool Equals(ProgrammaticSuppressionInfo other)
+        {
+            return other != null &&
```

Comment:
>return [](start = 12, length = 6)

Consider checking if this is the same instance before doing more expensive comparison.